### ISSUE TITLE
Metrics for time series references in shepard

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#133]
***In GitLab by @thaase on Aug 7, 2024, 09:47 GMT+2:***

# Feature Request

## :goal: Goal

As a `data-driven researcher working with timeseries` I want to `see metrics for timeseries items within a timeseries reference` so that `I can get a quick overview of any timeseries`.

## :medal: Acceptance Criteria

1. For single timeseries items within timeseries references (within the timespan of the reference) the following metrics should be available for the user:
   1. Calculate min/max/mean/std deviation/median for numerical measurements
   2. Provide first value/last value
   3. Number of measurements
   4. Frequency of measurements (Number of measurements divided by timespan of timeseries).
2. Metrics to a timeseries reference are available via rest endpoint
3. The metrics for the timeseries reference are displayed in the new timeseries-plotter-dialogue for an individual timeseries of a timeseries reference
4. The semantic annotations for single timeseries-items are displayed read only in the same dialogue

Design: https://www.figma.com/design/ivBPq71AuUl2ohFnrzHIKF/Visual-Design?node-id=2841-28406&m=dev

## :bulb: Hints

We shouldn't support this for custom timeseries with multiple value fields.

Follow up #131

## :thinking: Risks & Open Questions

* Open Q: How can we identify timeseries when retrieving metrics? --\> endpoint
* Open Q: Maybe have another place in the future where these measurements are displayed

**Assignees:** xit-mohamed

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/133*

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 7, 2024, 09:51 GMT+2:***

changed due date to August 07, 2024

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 7, 2024, 09:51 GMT+2:***

changed due date to March 31, 2025

**Comment by noheton** [comment]
***In GitLab by @thaase on Aug 7, 2024, 10:00 GMT+2:***

marked this issue as blocked by #131

**Comment by noheton** [comment]
***In GitLab by @thaase on Sep 5, 2024, 14:13 GMT+2:***

added to epic &14

**Comment by noheton** [comment]
Maybe have a look at https://docs.timescale.com/use-timescale/latest/continuous-aggregates/
A update policy might have to be discussed... 
Unfortunately I am not familiar with these, so it might be misleading.

**Comment by noheton** [comment]
***In GitLab by @wiglef on Mar 13, 2025, 09:21 GMT+1:***

unassigned @jessicafriedline

**Comment by noheton** [comment]
***In GitLab by @ZeroOne42 on Mar 20, 2025, 16:22 GMT+1:***

removed the relation with #131

**Comment by noheton** [comment]
***In GitLab by @wiglef on Apr 3, 2025, 11:13 GMT+2:***

@noheton  i just had a discussion with @ZeroOne42 on the following AC: 
- Metrics to a timeseries reference are available via rest endpoint 

to implement the planned frontend, we implement an endpoint to retreive the metrics for an individual timeseries within the timeseries reference due to performance considerations. question is now, if another endpoint to get all metrics for all timeseries within the reference is needed on top of that.

**Comment by noheton** [comment]
***In GitLab by @ZeroOne42 on Apr 3, 2025, 11:22 GMT+2:***

added #573 as child task

**Comment by noheton** [comment]
***In GitLab by @ZeroOne42 on Apr 3, 2025, 11:22 GMT+2:***

added #574 as child task

**Comment by noheton** [comment]
***In GitLab by @ZeroOne42 on Apr 3, 2025, 11:23 GMT+2:***

added #575 as child task

**Comment by noheton** [comment]
***In GitLab by @ZeroOne42 on Apr 4, 2025, 11:32 GMT+2:***

Alignment regarding REST endpoint for metrics with @noheton, @lettfe, @kauf_pt, @xit-eyad-aljawabreh, @xit-max-hey, @ZeroOne42 

* We create a rest endpoint that enables the user to request metrics for a single timeseries within a timeseries reference
* In order to identify a timeseries within the timeseries reference, we pass all timeseries fields (measurement, location, device, field, symbolic name) as query parameters
* In the midterm adressing a timeseries should be changed to use the timeseriesId like it is for timeseries containers. A new issue will be created and assigned to the newly created epic "Planned midterm development (DLR)"

**Comment by noheton** [comment]
***In GitLab by @jessicafriedline on Apr 10, 2025, 11:11 GMT+2:***

It looks good!! Some issues: 

*  I couldn't see the semantic annotation chips because I can't add them myself yet. 
* There are some minor UI things we need to do. Please give me a quick call, this should take only a few minutes
* A small bug: I realized that after I have clicked on the maximum number of time series I cannot deselect them anymore.

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 11, 2025, 09:41 GMT+2:***

mentioned in commit 53efe01d8eec253a3c7d431a2071afc170046b2b

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 11, 2025, 11:17 GMT+2:***

mentioned in commit 70181edb7ec90dae6e63f4fa8448826caf99efb9

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 11, 2025, 11:36 GMT+2:***

mentioned in commit 4bb0962369c462486f75fb73de19d229820e6575

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 11, 2025, 12:21 GMT+2:***

mentioned in commit fe9a6e6725f43889f85505eb01faa31db13ee28a

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 14, 2025, 07:53 GMT+2:***

Resolved and pushed!

**Comment by noheton** [comment]
***In GitLab by @xit-eyad-aljawabreh on Apr 14, 2025, 07:54 GMT+2:***

Ramaining comments:

* Left and down Arrow better to be changed to match the design.
* Bug: sorting timeseries table mutiple times increases number of entries.

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Apr 14, 2025, 14:31 GMT+2:***

mentioned in commit d7a36788e0a1084ef607f7d9905c80cae8b0b436

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Apr 14, 2025, 14:31 GMT+2:***

mentioned in commit 6a8e9176df8a0cd41dc639aa4ffe67e86d67801f

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Apr 14, 2025, 14:31 GMT+2:***

mentioned in commit b9e8d74d37478f65ce096e0483ae870031394c1a

**Comment by noheton** [comment]
***In GitLab by @xit-mohamed on Apr 23, 2025, 10:39 GMT+2:***

fixed
